migration/v0.43/apply/20-upgrade-opensearch.sh

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env bash
+
+ROOT="$(readlink -f "$(dirname "${0}")/../../../")"
+
+# shellcheck source=scripts/migration/lib.sh
+source "${ROOT}/scripts/migration/lib.sh"
+
+run() {
+  case "${1:-}" in
+  execute)
+
+    if [[ "${CK8S_CLUSTER}" =~ ^(sc|both)$ ]]; then
+      log_info "operation on service cluster"
+
+      if [[ ! "$(helm_chart_version "sc" opensearch-system opensearch-master)" = "2.26.1" ]]; then
+
+        user="admin"
+        password=$(sops --config "${CK8S_CONFIG_PATH}/.sops.yaml" -d "${CK8S_CONFIG_PATH}"/secrets.yaml | yq4 '.opensearch.adminPassword')
+        os_url=https://opensearch.$(yq4 '.global.opsDomain' "${CK8S_CONFIG_PATH}"/common-config.yaml)
+
+        resp=$(curl -sS -kL -u "${user}:${password}" -X GET "${os_url}"/.kibana_1)
+        if [[ $(echo "${resp}" | jq -r 'to_entries | .[0].key') == ".kibana_1" ]]; then
+
+          log_info "- Index .kibana_1 already exists, skipping"
+
+        elif [[ $(echo "${resp}" | jq -r '.error.type') == "index_not_found_exception" ]]; then
+
+          log_info "- Cloning index .opensearch_dashboards to .kibana"
+
+          log_info "- Getting name of .opensearch_dashboards index"
+          os_dashboards_index=$(curl -sS -kL -u "${user}:${password}" -X GET "${os_url}"/_alias/.opensearch_dashboards | jq -r 'to_entries | .[0].key')
+
+          if [[ "${os_dashboards_index}" != .opensearch_dashboards* ]]; then
+            log_fatal "Failed to get index name of the .opensearch_dashboards alias"
+          fi
+
+          log_info "- Marking index '${os_dashboards_index}' as read-only"
+          resp=$(curl -sS -kL -u "${user}:${password}" -X PUT "${os_url}"/"${os_dashboards_index}"/_settings -H 'Content-Type: application/json' -d'
+          {
+            "settings": {
+              "index.blocks.write": true
+            }
+          }
+          ')
+
+          acknowledged=$(echo "${resp}" | grep "^{" | jq -r '.acknowledged')
+          if [ "${acknowledged}" = "true" ]; then
+            log_info "- Marked '${os_dashboards_index}' as read-only"
+          else
+            log_fatal "Failed to mark index '${os_dashboards_index}' as read-only" "${resp}"
+          fi
+
+          log_info "- Cloning index '${os_dashboards_index}' to index .kibana_1"
+          resp=$(curl -sS -kL -u "${user}:${password}" -X PUT "${os_url}"/"${os_dashboards_index}"/_clone/.kibana_1)
+          acknowledged=$(echo "${resp}" | grep "^{" | jq -r '.acknowledged')
+          if [ "${acknowledged}" = "true" ]; then
+            log_info "- Successfully cloned '${os_dashboards_index}' to .kibana_1"
+          else
+            log_fatal "Failed to clone index '${os_dashboards_index}' to .kibana_1" "${resp}"
+          fi
+
+        else
+          log_fatal "Failed to check if index .kibana_1 already exists" "${resp}"
+        fi
```

Comment:
Maybe this can be moved to a separate function `clone_opensearch_dashboard_index`?